compiler/src/dotty/tools/dotc/quoted/Interpreter.scala

```diff
@@ -32,10 +32,15 @@ import dotty.tools.dotc.reporting.Message
 import dotty.tools.repl.AbstractFileClassLoader
 
 /** Tree interpreter for metaprogramming constructs */
-class Interpreter(pos: SrcPos, classLoader: ClassLoader)(using Context):
+class Interpreter(pos: SrcPos, classLoader0: ClassLoader)(using Context):
   import Interpreter._
   import tpd._
 
+  val classLoader =
+    if ctx.owner.topLevelClass.name.startsWith(str.REPL_SESSION_LINE) then
+        new AbstractFileClassLoader(ctx.settings.outputDir.value, classLoader0)
```

Comment:
Shouldn't the REPL take care of passing the correct classloader to the Interpreter?